### PR TITLE
Init

### PR DESCRIPTION
**Pull request by xh3b4sd**
Our libraries to manage Vault primitives are a bit messy. I try to restructure this to be useful for host cluster bootstrapping (via certctl), for certificate generation (via cluster-service) and for PKI backend management and certificate generation (via cert-operator). This lib intends to manage the PKI part. 

See also

- https://github.com/giantswarm/vaultcrt/pull/1
- https://github.com/giantswarm/vaultrole/pull/1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,3 @@
-# vaultpki
\ No newline at end of file
+[![CircleCI](https://circleci.com/gh/giantswarm/vaultpki.svg?&style=shield&circle-token=8adb655cf7ed6250c599312c2c613ba119280ad1)](https://circleci.com/gh/giantswarm/vaultpki)
+
+# vaultpki
```
Comment (rossf7) [inline README.md:3]: A short summary here would be good and maybe how it relates to the other vault libs.

File: key/key.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package key
```
Comment (rossf7) [inline key/key.go:1]: Big fan of the key package :)

**Review by rossf7 — APPROVED** [review]
I like that certctl is being restructured. Splitting into 3 is more repos but keeps them smaller so I'm onboard with that. A nitpick about more info in the README but otherwise LGTM

File: README.md (hunk 1 above)
Comment (xh3b4sd) [reply README.md:3]: Done.

**Review by kopiczko — APPROVED** [review]
I'm not sure about splitting methods of a single struct into multiple files, but go with it if it makes you happy. The code is simple anyway.

**Comment by xh3b4sd** [comment]
I am going with this now as a first step. The library was used by the refactored `cert-operator` for over a week now on `gauss`, `ginger` and `lycan`.

**Pull request merged**